nilearn/plotting/surf_plotting.py

```diff
@@ -3,18 +3,435 @@
+
+    Returns
+    =======
+    scipy.sparse.csr_matrix
+       Shape (n_voxels, n_mesh_vertices). The dot product of this matrix with
+       an image (represented as a column vector) gives the projection onto mesh
+       vertices.
+
+    """
+    mesh = load_surf_mesh(mesh)
+    sample_locations = _sample_locations(
+        mesh, affine, kind=kind, radius=radius, n_points=n_points)
+    sample_locations = np.asarray(np.round(sample_locations), dtype=int)
+    n_vertices, n_points, img_dim = sample_locations.shape
+    masked = _masked_indices(np.vstack(sample_locations), img_shape, mask=mask)
+    sample_locations = np.rollaxis(sample_locations, -1)
+    sample_indices = np.ravel_multi_index(
+        sample_locations, img_shape, mode='clip').ravel()
+    row_indices, _ = np.mgrid[:n_vertices, :n_points]
+    row_indices = row_indices.ravel()
+    row_indices = row_indices[~masked]
+    sample_indices = sample_indices[~masked]
+    weights = np.ones(len(row_indices))
+    proj = sparse.csr_matrix(
+        (weights, (row_indices, sample_indices.ravel())),
+        shape=(n_vertices, np.prod(img_shape)))
+    proj = sklearn.preprocessing.normalize(proj, axis=1, norm='l1')
+    return proj
+
+
+def _nearest_voxel_sampling(images, mesh, affine, kind='ball', radius=3.,
+                            n_points=None, mask=None):
+    """In each image, measure the intensity at each node of the mesh.
+
+    Image intensity at each sample point is that of the nearest voxel.
+    A 2-d array is returned, where each row corresponds to an image and each
+    column to a mesh vertex.
+    See documentation of niimg_to_surf_data for details.
+
+    """
+    proj = projection_matrix(
+        mesh, affine, images[0].shape, kind=kind, radius=radius,
+        n_points=n_points, mask=mask)
+    data = np.asarray(images).reshape(len(images), -1).T
+    texture = proj.dot(data)
+    texture[np.asarray(proj.sum(axis=1) == 0).ravel()] = np.nan
+    return texture.T
+
+
+def _interpolation_sampling(images, mesh, affine, kind='ball', radius=3,
+                            n_points=None, mask=None):
+    """In each image, measure the intensity at each node of the mesh.
+
+    Image intensity at each sample point is computed with trilinear
+    interpolation.
+    A 2-d array is returned, where each row corresponds to an image and each
+    column to a mesh vertex.
+    See documentation of niimg_to_surf_data for details.
+
+    """
+    sample_locations = _sample_locations(
+        mesh, affine, kind=kind, radius=radius, n_points=n_points)
+    n_vertices, n_points, img_dim = sample_locations.shape
+    grid = [np.arange(size) for size in images[0].shape]
+    interp_locations = np.vstack(sample_locations)
+    masked = _masked_indices(interp_locations, images[0].shape, mask=mask)
+    # loop over images rather than building a big array to use less memory
+    all_samples = []
+    for image in images:
+        interpolator = interpolate.RegularGridInterpolator(
```

Comment:
Seems like this is new from SciPy 0.14.x. Do you know if there any alternatives for < 0.14 SciPy version. Because we support 0.13.3

Reply:
Thanks for pointing this out. I'm not sure that there is an easy alternative,
because interpolation of data on a regular grid was introduced in 0.14. Before
that, the linear interpolation accepts a bunch of arbitrary points and computes
a delaunay triangulation, which is prohibitively slow - I waited 10 mn for the
2mm MNI template before giving up.
We could:
  - write our own linear interpolation, which would give us the opportunity to
    present it as a sparse matrix, with the same advantages we got from doing
    this for the nearest voxel interpolation.
  - issue a warning and fall back to nearest neighbour interpolation when the
    scipy version is < 0.14. Since the resulting images are almost identical
    (at least qualitatively, i.e. the plots will look the same), this seems
    reasonable to me.

What do you think?

Reply:
>What do you think?

2 option looks reasonable compromise. So, here nearest neighbour is faster compared to linear one ?

Reply:
nearest neighbour does not use scipy.interpolate so it does not have this issue.